### ISSUE TITLE
Velocity not syncing datapack registries correctly, affecting modded fabric platforms

### ISSUE THREAD

**Issue by purejosh** [PaperMC/Velocity#922]
In the example of Supplementaries, players are kicked when using items that call to a specific api. I've reported this to the developers, and they're stating that velocity isn't handling the sync of datapack registries correctly. 

Here's the client crash that's happening: 
https://gist.github.com/purejosh/c6da9ecba61745eb4ccf69737ea2fc25

![image](https://user-images.githubusercontent.com/70633949/210854745-1689cd2a-d354-472c-883e-12f361d72046.png)
 
Please let me know what other info you may need. Might be easier to start an open conversation with the dev of Supplementaries, MehVahdJukaar.

**Comment by astei** [comment]
I looked into it and couldn't see anything obviously wrong. I've committed 022ae0fc002800b42f6b870e62cf52d5a7cb5341 to provide better syncing around some of the registries that Velocity is forced to intercept (by simply passing it through), and this may resolve the issue, but past that I'm not sure what else we can do.

**Comment by SanderWxD** [comment]
Yep, same issue here. Sadly it doesn't seem like the above mentioned commit resolved the issue. Velocity/Paper is now not compatible with popular modpacks like FTB Direwolf20 1.20.2. Clients crash instantly upon joining.

[crash-2023-11-27_20.26.01-client.txt](https://github.com/PaperMC/Velocity/files/13480504/crash-2023-11-27_20.26.01-client.txt)

**Comment by electronicboy** [comment]
The problem here is likely going to be down to the whole login phase snafu, in which there is generally little we can do given that we cannot reset the client back into the login phase, which is going to cause issues with stuff like forge. There is the ambassador plugin (which requires a mod) in order to try to facilitate this, ofc, usual caveats around what the client expects from the server apply (not all mods will allow you to join a "vanilla" server, be that mojangs software or something like paper)
